### PR TITLE
fix(errors): error abstractions refactoring 

### PR DESCRIPTION
**Pull request by jsmvaldivia**
After working with the code a bit I noticed the cause for both the new exception are different so I decided to split the previously discussed error into 2: one is related to a OPA loader error and we just manage to wrap it and the other is actually related to the policy files.
Since I actually found the `Filtered` method for the loader a quite clean solution, I decided to keep it and extracted the actual predicate for better readability. This implies that we cannot find out what are the files as they might have been filtered out but I believe that after inlining the `allRepos()` function it reads a bit better than before and we do not obfuscate the errors purposes as much.
In terms of the NoPolicyError found I rephrased its message to `no policy .rego files found in "path"` as it no longer implies that there are no policy files but rather that there aren't no .rego policy files that might help better indicate any issue.

### CODE REVIEW

File: internal/policy/errors.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package policy
+
+import "fmt"
+
+type PolicyLoaderError struct {
+	loaderError error
+}
+
+func (e *PolicyLoaderError) Error() string {
```
Comment (crqra) [inline internal/policy/errors.go:9]: I think it's better that the errors names are like `ErrSomethingWentWrong`, like some (if not all) packages from the stdlib do. It's easier to look for them IMHO. What do you think?

```suggestion
type ErrPolicyLoad struct {
	loaderError error
}

func (e *ErrPolicyLoad) Error() string {
```

File: internal/policy/errors.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	return fmt.Sprintf("load: %v", e.loaderError)
+}
+
+type NoPoliciesError struct {
+	policyPaths []string
+}
+
+func (e *NoPoliciesError) Error() string {
```
Comment (crqra) [inline internal/policy/errors.go:17]: Same as above.

```suggestion
type ErrNoPolicies struct {
	policyPaths []string
}

func (e *ErrNoPolicies) Error() string {
```

**Review by crqra — CHANGES_REQUESTED** [review]
Hey @jsmvaldivia 👋🏼 

I think it was a good approach splitting the errors into two and changing predicate to a separate function, it's more clear that way.

Just left a couple of suggestions regarding the errors names, let me know your thoughts. Also, I don't think we've any real benefit of having a separate file for the errors they could probably live fine inside the `engine.go` file in their context, but I leave that for you to decide what is preferred.

Thanks!! ❤️

File: internal/policy/errors.go (hunk 1 above)
Comment (jsmvaldivia) [reply internal/policy/errors.go:9]: Still picking up these standards. Will change!

File: internal/policy/errors.go (hunk 2 above)
Comment (jsmvaldivia) [reply internal/policy/errors.go:17]: Changing!

**Comment by jsmvaldivia** [comment]
@crqra Did the changes as requested.
Regarding the errors.go file, I see what you mean, I believe we can leave these files a bit smaller. In the end, they're still in the same go package.  If anything we could easily restructure this if needed but I leave that decision to you

**Review by crqra — APPROVED** [review]
LGTM 👍

**Pull request merged**